internal/pkg/aws/sessions/sessions.go

```diff
@@ -59,6 +63,10 @@ func (p *Provider) Default() (*session.Session, error) {
 	if err != nil {
 		return nil, err
 	}
+	if aws.StringValue(sess.Config.Region) == "" {
+		return sess, ErrMissingRegion
```

Comment:
hmmm, this is a bit unusual. The session itself is valid if the error is `ErrMissingRegion`. This breaks the convention of the first return argument being the zero value if there is an error, and we end up using the first arg in the `FromRole` method.

What do you think of instead:
1. This method should return `nil, ErrMissingRegion`.
2. In `FromRole` instead we don't call `p.Default` and instead write separately:
```
sess, err := session.NewSessionWithOptions(session.Options{
		Config:            *newConfig(),
		SharedConfigState: session.SharedConfigEnable,
})
```